### PR TITLE
Log error when the return from the remote method leads to an error

### PR DESCRIPTION
**Pull request by ayeshLK**
Fixes [#1449](https://github.com/ballerina-platform/ballerina-standard-library/issues/1449)

## Checklist
- [X] Linked to an issue
- [X] Updated the changelog
~~Added tests~~ N/A

### CODE REVIEW

File: websubhub-native/src/main/java/io/ballerina/stdlib/websubhub/HubCallback.java [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+            if (!isModuleDefinedError(error)) {
+                error.printStackTrace();
+            }
+        }
+
+        future.complete(result);
+    }
+
+    @Override
+    public void notifyFailure(BError bError) {
```
Comment (chamil321) [inline websubhub-native/src/main/java/io/ballerina/stdlib/websubhub/HubCallback.java:56]: We have to log the error in this method as well. Basically `checkpanic` comes here

File: websubhub-native/src/main/java/io/ballerina/stdlib/websubhub/HubCallback.java [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+        future.complete(invocationError);
+    }
+
+    private boolean isModuleDefinedError(BError error) {
+        Type errorType = error.getType();
+        String errorName = errorType.getName();
+        Module packageDetails = errorType.getPackage();
+        String orgName = packageDetails.getOrg();
+        String packageName = packageDetails.getName();
+        return Constants.MODULE_DEFINED_ERRORS.contains(errorName)
```
Comment (chamil321) [inline websubhub-native/src/main/java/io/ballerina/stdlib/websubhub/HubCallback.java:69]: Won't the `PACKAGE_NAME` and `PACKAGE_ORG` be enough to filter module errors as only the error types reach here?

File: websubhub-native/src/main/java/io/ballerina/stdlib/websubhub/NativeHttpToWebsubhubAdaptor.java [hunk 3]
```diff
@@ -21,34 +21,27 @@
...
-
 /**
  * {@code NativeHttpToWebsubhubAdaptor} is a wrapper object used for service method execution.
  */
 public class NativeHttpToWebsubhubAdaptor {
-    private static final String SERVICE_OBJECT = "WEBSUBHUB_SERVICE_OBJECT";
-
     public static void externInit(BObject adaptor, BObject serviceObj) {
-        adaptor.addNativeData(SERVICE_OBJECT, serviceObj);
+        adaptor.addNativeData(Constants.SERVICE_OBJECT, serviceObj);
```
Comment (chamil321) [inline websubhub-native/src/main/java/io/ballerina/stdlib/websubhub/NativeHttpToWebsubhubAdaptor.java:40]: Shall we use static import for `SERVICE_OBJECT` as it's quite redundant?

File: changelog.md [hunk 4]
```diff
@@ -7,6 +7,8 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 ## [Unreleased]
 - Multipart content delivery using websubhub-client.
```
Comment (shafreenAnfar) [inline changelog.md:8]: I think this is incorrect. We don't need add unplanned tasks under unreleased section. Unrelease section is basically there to say what to expect in the coming release. Also unreleased section should shave same structure as added, removed, changed etc.

**Review by chamil321 — APPROVED** [review]

**Review by chamil321 — APPROVED** [review]

**Pull request merged**